src/ddsrt/src/events/posix/events_posix.c

```diff
@@ -75,102 +77,79 @@ static dds_return_t ddsrt_event_queue_init(ddsrt_event_queue_t* queue) {
   addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   addr.sin_port = 0;
   if (bind(listener, (struct sockaddr*)&addr, sizeof(addr)) == -1 ||
-    getsockname(listener, (struct sockaddr*)&addr, &asize) == -1 ||
-    listen(listener, 1) == -1 ||
-    connect(queue->interrupt[0], (struct sockaddr*)&addr, sizeof(addr)) == -1 ||
-    (queue->interrupt[1] = accept(listener, 0, 0)) == -1) {
-    closesocket(queue->interrupt[0]);
-    closesocket(queue->interrupt[1]);
-    return DDS_RETCODE_ERROR;
-  }
+      getsockname(listener, (struct sockaddr*)&addr, &asize) == -1 ||
+      listen(listener, 1) == -1 ||
+      connect(queue->interrupt[0], (struct sockaddr*)&addr, sizeof(addr)) == -1 ||
+      (queue->interrupt[1] = accept(listener, 0, 0)) == -1)
+    goto pipe_cleanup;
+  closesocket(listener);
   SetHandleInformation((HANDLE)queue->interrupt[0], HANDLE_FLAG_INHERIT, 0);
   SetHandleInformation((HANDLE)queue->interrupt[1], HANDLE_FLAG_INHERIT, 0);
-#elif defined(__VXWORKS__)  
-  /*vxworks type pipe*/
-  char pipename[OSPL_PIPENAMESIZE];
-  int pipecount = 0;
-  int pipe_result = 0;
-  do {
-    snprintf((char*)&pipename, sizeof(pipename), "/pipe/ospl%d", pipecount++);
-  } while ((pipe_result = pipeDevCreate((char*)&pipename, 1, 1)) == -1 &&
-    os_getErrno() == EINVAL);
-  if (pipe_result != -1)
-    return DDS_RETCODE_ERROR;
-  queue->interrupt[0] = open((char*)&pipename, O_RDWR, 0644);
-  queue->interrupt[1] = open((char*)&pipename, O_RDWR, 0644);
-  /*the pipe was succesfully created, but one of the sockets on either end was not*/
-  if (-1 == queue->interrupt[0] || -1 == p[1]) {
-    pipeDevDelete(pipename, 0);
-    if (-1 != queue->interrupt[0])
-      close(queue->interrupt[0]);
-    if (-1 != queue->interrupt[1])
-      close(queue->interrupt[1]);
-    return DDS_RETCODE_ERROR;
-  }
 #elif !defined(LWIP_SOCKET)
   /*simple linux type pipe*/
-  if (pipe(queue->interrupt) == -1) return DDS_RETCODE_ERROR;
-#endif /* _WIN32, __VXWORKS__, !LWIP_SOCKET*/
+  if (pipe(queue->interrupt) == -1)
+    goto alloc_cleanup;
+#endif /* _WIN32 || !LWIP_SOCKET*/
+  ddsrt_mutex_init(&queue->lock);
```

Comment:
here (like in the `kqueue` variant), making the "interrupt" pipe non-blocking and close-on-exec would make sense.

Reply:
I have the pipe non-blocking.
I am not sure what you mean with making the pipe close-on-exec, since there is no child process.

Reply:
The point is that the application may fork and create child processes.